tests/integration/security/util/framework.go

```diff
@@ -48,4 +48,10 @@ func EchoConfig(name string, ns namespace.Instance, headless bool, annos echo.An
 		Galley: g,
 		Pilot:  p,
 	}
+
+	// for headless service, the port and target port must be equal
+	if headless {
```

Comment:
how does this affect the inbound listener config generated on the headless server side? before and after?

Reply:
It does not influence server-side inbound config. Just function on outbound direction, as the headless service is ORIGINAL_DST, if the ports are different, the connection to podip:port will not be accepted.

Reply:
Now I understand this better, when using the service port, then traffic is going through

listener, 0.0.0.0_<svc-port> -> outbound||headless-svc|port|cluster  ----- > server app listener (with TLS context specified), it works.

when using target port, <pod-ip>:<port>, then it going through the ORIGIN_DST cluster, thus no client's tls config, fails server side tls.

so when using <pod-ip>:<target-port>, it works under plain text, but not mtls.